### ISSUE TITLE
[3.2.4 beta 2] Bug with the focus remaining on the main menu

### ISSUE THREAD

**Issue by dalexeev** [godotengine/godot#43695]
**Godot version:**
3.2.3 - OK
3.2.4.beta2 - Bug

**OS/device including version:**
Kubuntu 20.10

**Issue description:**
3.2.3:
![3.2.3](https://user-images.githubusercontent.com/47700418/99690323-107e2f00-2a99-11eb-8d0e-3130093920f0.gif)

3.2.4.beta2:
![3.2.4.beta2](https://user-images.githubusercontent.com/47700418/99690338-14aa4c80-2a99-11eb-8820-2d31bb0d3341.gif)

**Steps to reproduce:**
See above.

**Minimal reproduction project:**
—

**Comment by akien-mga** [comment]
Confirmed, the issue also happens with 3.2.4.beta1, so the regression happened before beta1 (i.e. before 2e073ecbeaf5b502c2b8c3c0510e4a22a56db58f).

**Comment by akien-mga** [comment]
That seems to be a regression from #41577, CC @FIF15. We might have overlooked something in that PR.

**Comment by FIF15** [comment]
> That seems to be a regression from #41577, CC @FIF15. We might have overlooked something in that PR.

I've confirmed that the removal of `set_enabled_focus_mode(FOCUS_NONE);` from #41577 is the direct reason for this bug. We removed it because the default focus mode of Controls is FOCUS_NONE, however, it is set to FOCUS_ALL in the BaseButton's constructor. Therefore, the removal changed the default behavior of MenuButton and LinkButton. Also, this shall happen on the master. I would make PR for both 3.2 and master to fix this.
